music21/environment.py

```diff
@@ -1566,31 +1583,34 @@ def testEnvironmentA(self):
         self.assertEqual(list(env['localCorpusSettings']), ['/a', '/b'])
 
     @unittest.skipUnless(
-        os.access(Environment().getDefaultRootTempDir(), stat.S_IRWXU),
-        'test will programmatically set read/write/exec permissions on this dir'
+        common.getPlatform() in ['nix', 'darwin'],
+        'os.getuid can be called only on Unix platforms'
     )
-    @unittest.skipIf(
-        common.getPlatform() == 'win',
-        'os.chmod does not have the intended effect on Windows'
-    )
-    def testGetTempFile(self):
-        import getpass
+    def testGetDefaultRootTempDir(self):
         import stat
+        import shutil
 
         e = Environment()
         oldScratchDir = e['directoryScratch']
         try:
             e['directoryScratch'] = None
+            oldTempDir = e.getDefaultRootTempDir()
+            oldPermission = oldTempDir.stat()[stat.ST_MODE]
             # Wipe out write, exec permissions on the default root dir
-            os.chmod(e.getDefaultRootTempDir(), stat.S_IREAD)
-            # Was the PermissionError caught and a new "music21-{user}" dir created?
-            tmp = e.getTempFile(returnPathlib=False)
-            self.assertIn('music21-' + getpass.getuser(), tmp)
+            os.chmod(oldTempDir, stat.S_IREAD)
+            newTempDir = e.getDefaultRootTempDir()
+            self.assertIn(f'music21-userid-{os.getuid()}', str(newTempDir))
         finally:
-            # Restore owner read/write/exec permissions and original path
-            os.chmod(e.getDefaultRootTempDir(), stat.S_IRWXU)
+            # Restore original permissions and original path
+            os.chmod(oldTempDir, oldPermission)
             e['directoryScratch'] = oldScratchDir
-        os.remove(tmp)
+
+            # If getting OSError while trying to create the directory on the first fallback,
+            # the default temp directory from tempfile.gettempdir() will be return on the second
+            # fallback. We don't want to delete the default temp directory. Therefore we check
+            # it before deleting.
+            if not newTempDir.samefile(tempfile.gettempdir()):
```

Comment:
because this is in `finally` code, cannot assume that `newTempDir` is defined at this point.  Define it as None outside the try and then test to see if it is not None first before running samefile.  Should also check that it is not the same file as 'directoryScratch' or oldScratchDir before running.